articles/express/index.md

```diff
@@ -0,0 +1,318 @@
+$ touch static/dummy_file.txt
+$ touch static/dummy_file2.txt
+$ echo file1 > static/dummy_file.txt
+$ echo file2 > static/dummy_file2.txt
+$ touch app.js
+```
+
+Our app will have a [logger](https://en.wikipedia.org/wiki/Log_file) function and a static file serving function.
+
+
+```JavaScript
+//app.js
+
+var express = require("express");
+var path = require("path");
+var fs = require("fs");
+
+var app = express();
+
+
+//1. Logging
+app.use(function(req, res, next) {
+    console.log("Request IP: " + req.url);
+    console.log("Request date: " + new Date());
+    next();
+});
+
+
+//2. File Server
+app.use(function(req, res, next) {
+    var filePath = path.join(__dirname, "static", req.url);
+    fs.stat(filePath, function(err, fileInfo) {
+    if (err) {
+        next();
+        return;
+    }
+    if (fileInfo.isFile()) {
+        res.sendFile(filePath);
+    } else {
+        next();
+    }
+  });
+});
+
+
+app.listen(3000, function() {
+    console.log("App started on port 3000");
+});
+
+```
+
+If we run this file using `node app.js` and go to `localhost:3000/dummy_file.txt`, we can see on the screen `file1`. If we go to the URL `localhost:3000`, we see an error `Cannot GET /` because we did not configure a route handler for that path. Let's look at the code.
+
+The logger logs every request that comes into the server. `app.use` is used to define a middleware function, it takes a function. The `next()` function call tells Express to move onto the next function in the stack (remove the `next()` call in your script, you will notice that it takes forever for the page to load, this is because the request gets stuck on this middleware function).
+
+We are using the [path module](https://nodejs.org/api/path.html) to join the relative URL (from the request) and the directory name. The [fs module](https://nodejs.org/api/fs.html) provides an API for interacting with the file system. We are checking if the file exists, if it does not, we will go to next function in the stack if it does we will return that file using `res.sendFile`.
+
+
+#### Using Third-Party Middleware
+We can write our own middleware functions or import them similar to how we imported our modules in Node.js using `require`. Let's use a popular open-source logger called [morgan](http://expressjs.com/en/resources/middleware/morgan.html) instead of writing our own logging function.
+
+Install it using npm.
+
+```bash
+$ npm install morgan
+```
+
+We can call the `use()` on the Express app object to add the middleware to the stack.
+
+```JavaScript
+
+var express = require("express");
+var logger_morgan = require("morgan");
+
+var app = express();
+...
+app.use(logger_morgan("short")); // logs short notation of requests
+...
+```
+
+Express comes with **`express.static`** middleware bundled with it, it can be used to serve static files instead of the function in the previous section. It provides better security and performance than the function we wrote.
+
+`JavaScript
```

Comment:
This code block should be surrounded by three backticks. Please correct other examples.

Reply:
I think it was changed during preliminary review by @hectorkambow . I always type `javascript` but I see `JavaScript` now.